templates/_worker.js/utils/routing.ts

```diff
@@ -70,8 +70,10 @@ export async function runOrFetchBuildOutputItem(
 
 	try {
 		switch (item?.type) {
-			case 'function': {
-				resp = await (await item.entrypoint).default(req, ctx);
+			case 'function':
+			case 'middleware': {
+				const edgeFunction: EdgeFunction = await import(item.entrypoint);
```

Comment:
I checked with `wrangler dev --experimental-local`, and this **does not** re-evaluate the module every time. It only runs evaluates it the first time it gets imported. This is good!